### ISSUE TITLE
Add config option to set is_approved=True for private sites

### ISSUE THREAD

**Issue by johntwyman** [ealgis/ealgis#197]
Hi team,

Wondering if we could look to support the idea of setting `is_approved` = `True` for private EALGIS installations on the basis of a configuration option. I was thinking an environment variable like `EALGIS_PRIVATE_SITE_PREAPPROVE` or similar.

Our custom OAuth provider lets us set (and revoke) application level access in our Identity Provider. It's effectively a redundant step to then login to EALGIS and set the Profile to "Approved".

I'm mindful that this could go horribly wrong. A private site with this flag inadvertently set isn't private. Perhaps a necessary inclusion is a test of whether a custom OAuth provider is the source of the new user request?

What do you think?

(Also, while I'm no Python/Django coder, I'd be happy to have a crack at producing a "first pass" PR for this. But I'd rather we have the conversation first.)

**Comment by keithamoss** [comment]
Hey John, I agree - I think a change like this makes a lot of sense in the context of Private Site + Custom OAuth provider that's already managing approvals.

I think the best place to insert it would be in python-social-auth's auth pipeline for registering new users. There's some doco over here about [Extending the pipeline](https://python-social-auth-docs.readthedocs.io/en/latest/pipeline.html#extending-the-pipeline) that it would pay to look over.

You might want to look over something similar I wrote for another project where I've got a custom action at the end of python-social-auth's auth pipeline for getting user avatars from social auth services.

In short:

1. Define a function with your custom behaviour: https://github.com/keithamoss/scremsong/blob/master/django/scremsong/app/auth.py#L37-L54
2. Add it to the end of the `SOCIAL_AUTH_PIPELINE`: https://github.com/keithamoss/scremsong/blob/master/django/scremsong/settings.py#L143

PRs very happily accepted :)
